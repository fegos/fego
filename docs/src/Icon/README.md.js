webpackJsonp([12],{466:function(t,n){t.exports={content:["article",["h1","\u56fe\u6807 Icon"],["h3","\u7ec4\u4ef6\u63cf\u8ff0"],["ul",["li",["p","\u5e38\u7528\u56fe\u6807\u96c6\u5408"]]],["h3","\u793a\u4f8b\u4ee3\u7801"],["pre",{lang:"html",highlighted:'<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Icon</span> <span class="token attr-name">name</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>success<span class="token punctuation">"</span></span> <span class="token punctuation">/></span></span>\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Icon</span> <span class="token attr-name">name</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>error<span class="token punctuation">"</span></span> <span class="token attr-name">color</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>red<span class="token punctuation">"</span></span> <span class="token attr-name">size</span><span class="token attr-value"><span class="token punctuation">=</span>{38}</span> <span class="token punctuation">/></span></span>'},["code",'<Icon name="success" />\n<Icon name="error" color="red" size={38} />']],["h3","\u8bf4\u660e"],["ul",["li",["p","\u6240\u6709\u56fe\u6807\u5747\u6258\u7ba1\u5728 ",["a",{title:null,href:"http://iconfont.cn/"},"iconfont"]]],["li",["p","\u66f4\u65b0\u56fe\u6807\u5e93\u65f6\u9700\u5c06\u6587\u4ef6\u4e0b\u8f7d\u5230\u672c\u5730\uff0c\u7136\u540e\u7528 iconfont.js \u6587\u4ef6\u91cc\u7684\u5185\u5bb9\u66ff\u6362 Icon \u7ec4\u4ef6\u76ee\u5f55\u4e0b font.js \u6587\u4ef6\u91cc\u7684\u4e3b\u5185\u5bb9"]],["li",["p","\u5173\u4e8e\u56fe\u6807\u7684\u66f4\u591a\u7528\u6cd5\u53c2\u89c1 iconfont \u7684",["a",{title:null,href:"http://iconfont.cn/help/detail?spm=a313x.7781069.1998910419.d0091c141&helptype=code"},"\u5b98\u65b9\u8bf4\u660e"]]]],["h2","API"],["h3","props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","className"],["td","\u6837\u5f0f\u7c7b\u540d\uff0c\u53ef\u7528\u4e8e\u8986\u76d6\u9ed8\u8ba4\u6837\u5f0f"],["td","string"],["td"]],["tr",["td","name"],["td","\u56fe\u6807\u540d"],["td","string"],["td"]],["tr",["td","spin"],["td","\u56fe\u6807\u662f\u5426\u65cb\u8f6c"],["td","boolean"],["td","false"]],["tr",["td","width"],["td","\u56fe\u6807\u5bbd"],["td","number"],["td"]],["tr",["td","height"],["td","\u56fe\u6807\u9ad8"],["td","number"],["td"]],["tr",["td","size"],["td","\u56fe\u6807\u5c3a\u5bf8\uff0c\u4f7f\u7528\u65f6\u4f1a\u5bfc\u81f4 width \u548c height \u5931\u6548"],["td","number"],["td"]],["tr",["td","style"],["td","\u5355\u72ec\u6307\u5b9a\u56fe\u6807\u6837\u5f0f"],["td","style"],["td"]],["tr",["td","color"],["td","\u5355\u72ec\u6307\u5b9a\u56fe\u6807\u989c\u8272"],["td","string"],["td"]]]]],meta:{title:"Icon",subTitle:"\u56fe\u6807",filename:"src/Icon/README.md"},toc:["ul",["li",["a",{className:"bisheng-toc-h1",href:"#\u56fe\u6807-Icon",title:"\u56fe\u6807 Icon"},"\u56fe\u6807 Icon"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u7ec4\u4ef6\u63cf\u8ff0",title:"\u7ec4\u4ef6\u63cf\u8ff0"},"\u7ec4\u4ef6\u63cf\u8ff0"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u793a\u4f8b\u4ee3\u7801",title:"\u793a\u4f8b\u4ee3\u7801"},"\u793a\u4f8b\u4ee3\u7801"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u8bf4\u660e",title:"\u8bf4\u660e"},"\u8bf4\u660e"]],["li",["a",{className:"bisheng-toc-h2",href:"#API",title:"API"},"API"]],["li",["a",{className:"bisheng-toc-h3",href:"#props\u5217\u8868",title:"props\u5217\u8868"},"props\u5217\u8868"]]]}}});