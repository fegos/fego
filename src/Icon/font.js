// 防止node报错
if(typeof window !== 'undefined'){

	(function(window){var svgSprite='<svg><symbol id="nsicon-home" viewBox="0 0 1025 1024"><path d="M1017.395271 622.848l-452.0448-499.6096c-14.0288-15.5136-32.9728-24.064-53.3504-24.064 0 0 0 0 0 0-20.3264 0-39.2704 8.5504-53.3504 24.064l-452.0448 499.6096c-9.472 10.496-8.6528 26.6752 1.792 36.1472 4.9152 4.4544 11.0592 6.6048 17.152 6.6048 6.9632 0 13.9264-2.816 18.9952-8.448l109.0048-120.4736 0 410.5216c0 42.3424 34.4576 76.8 76.8 76.8l563.2 0c42.3424 0 76.8-34.4576 76.8-76.8l0-410.5216 109.0048 120.4736c9.472 10.496 25.6512 11.3152 36.1472 1.792s11.3152-25.6512 1.792-36.1472zM614.400071 972.8l-204.8 0 0-230.4c0-14.1312 11.4688-25.6 25.6-25.6l153.6 0c14.1312 0 25.6 11.4688 25.6 25.6l0 230.4zM819.200071 947.2c0 14.1312-11.4688 25.6-25.6 25.6l-128 0 0-230.4c0-42.3424-34.4576-76.8-76.8-76.8l-153.6 0c-42.3424 0-76.8 34.4576-76.8 76.8l0 230.4-128 0c-14.1312 0-25.6-11.4688-25.6-25.6l0-467.0976 291.84-322.56c4.1984-4.6592 9.6768-7.2192 15.36-7.2192s11.1616 2.56 15.36 7.2192l291.84 322.56 0 467.0976z"  ></path></symbol><symbol id="nsicon-success" viewBox="0 0 1024 1024"><path d="M510.7311008 35.89137387C248.50165547 35.89137387 35.92466453 248.46836373 35.92466453 510.6967168c0 262.21962133 212.57699093 474.7966112 474.8064352 474.7966112s474.8064352-212.57699093 474.8064352-474.7966112C985.537536 248.46836373 772.96054507 35.89137387 510.7311008 35.89137387L510.7311008 35.89137387 510.7311008 35.89137387zM796.7077984 377.93870187L468.16592853 713.5864096c-1.12645547 1.68422507-2.37188693 3.2844032-3.837808 4.77324587-12.87892267 13.0775808-33.75436907 13.0775808-46.6376576 0l-163.19960106-165.70465494c-12.86800747-13.08303787-12.86800747-34.27175253 0-47.3220448 12.87237333-13.078672 33.75436907-13.078672 46.6267424 0l139.39777173 141.50332694 309.55039787-316.25564587c12.8876544-13.07648853 33.75327787-13.07648853 46.6365664 0 12.86800747 13.08194667 12.86800747 34.2761184 0 47.35915627L796.7077984 377.93979307 796.7077984 377.93870187zM796.7077984 377.93870187"  ></path></symbol><symbol id="nsicon-warning" viewBox="0 0 1024 1024"><path d="M512 34.13333333C248.14933333 34.13333333 34.13333333 248.14933333 34.13333333 512c0 263.91893333 214.016 477.86666667 477.86666667 477.86666667 263.9872 0 477.86666667-213.94773333 477.86666667-477.86666667C989.86666667 248.14933333 775.9872 34.13333333 512 34.13333333zM512 785.06666667c-28.19413333 0-51.2-22.9376-51.2-51.2S483.80586667 682.66666667 512 682.66666667c28.39893333 0 51.2 22.9376 51.2 51.2S540.39893333 785.06666667 512 785.06666667zM563.2 529.06666667C563.2 557.39733333 540.39893333 580.26666667 512 580.26666667 483.80586667 580.26666667 460.8 557.39733333 460.8 529.06666667l0-238.93333334C460.8 261.80266667 483.80586667 238.93333333 512 238.93333333c28.39893333 0 51.2 22.86933333 51.2 51.2L563.2 529.06666667z"  ></path></symbol><symbol id="nsicon-question-fill" viewBox="0 0 1024 1024"><path d="M522.8639648 30.2525472c-265.00192213 0-480.5817024 215.57978027-480.5817024 480.58279467 0 265.0139296 215.57978027 480.59370987 480.5817024 480.59370986 265.0150208 0 480.59480107-215.57978027 480.59480107-480.59370986C1003.45876587 245.83232747 787.87898453 30.2525472 522.8639648 30.2525472L522.8639648 30.2525472 522.8639648 30.2525472zM522.8639648 854.48939627c-28.415712 0-51.48512747-23.06286613-51.48512747-51.49058454 0-28.415712 23.06941547-51.48512747 51.48512747-51.48512746 28.42771947 0 51.49713387 23.06941547 51.49713387 51.48512746C574.36109867 831.4265312 551.29714133 854.48939627 522.8639648 854.48939627L522.8639648 854.48939627 522.8639648 854.48939627zM617.88899627 503.52320427c-29.86635093 29.83688-60.68888 60.66595733-60.68888 88.35907946l0 58.05393387c0 18.94781013-15.38179307 34.32414613-34.33615254 34.32414613-18.9401696 0-34.32305387-15.37633493-34.32305386-34.32414613L488.54090987 591.88228373c0-56.1557696 42.91336853-99.06368107 80.8078976-136.92764693 27.76952853-27.76952853 56.50178347-56.47231253 56.50178346-79.36926613 0-57.2254656-46.20432107-103.76925227-102.9877184-103.76925227-57.73084267 0-102.97352853 44.55611627-102.97352853 101.43993387 0 18.94781013-15.38179307 34.32196267-34.33069547 34.32196266-18.95217707 0-34.328512-15.37415253-34.328512-34.32196266 0-93.7883328 76.9973792-170.09695787 171.63819307-170.09695787 94.6429984 0 171.64037653 77.34339307 171.64037653 172.42627627C694.50434133 426.9384224 655.54557333 465.86771947 617.88899627 503.52320427L617.88899627 503.52320427 617.88899627 503.52320427zM617.88899627 503.52320427"  ></path></symbol><symbol id="nsicon-question-o" viewBox="0 0 1024 1024"><path d="M1013.732 504.692c0.426 271.101-218.847 490.81-490.33 491.291-271.057 0.481-490.745-218.781-491.291-490.33-0.535-270.991 218.835-490.799 490.33-491.291 271.035-0.492 490.853 218.901 491.291 490.33zM962.308 505.599c0.95-241.412-194.697-437.661-438.458-439.812-240.889-2.13-439.408 195.789-440.315 438.96-0.907 241.468 194.751 437.715 438.458 439.812 240.943 2.076 439.354-195.724 440.315-438.96z"  ></path><path d="M548.001 665.387c-21.147 0-42.533 0-64.979 0-1.747-27.11 0.219-53.467 10.016-78.993 12.67-33.008 36.493-57.638 60.992-81.789 16.995-16.755 34.646-32.91 50.932-50.332 30.671-32.833 37.563-83.755 17.793-124.147-10.585-21.627-27.536-36.526-50.036-44.291-30.442-10.518-61.582-11.043-92.144-0.983-42.763 14.079-62.511 47.405-70.615 89.184-2.468 12.747-3.168 25.843-4.73 39.234-21.299 0-42.347 0-64.192 0-1.3-21.769 1.387-42.642 7.056-63.002 22.031-79.081 84.978-127.577 167.161-129.772 37.104-0.994 73.313 2.458 106.704 20.174 48.137 25.548 73.651 66.322 78.556 120.149 3.648 39.988-2.993 77.868-26.411 111.411-5.713 8.181-12.922 15.488-20.228 22.358-19.366 18.197-39.213 35.87-58.764 53.859-30.55 28.115-47.055 62.434-45.831 104.53 0.098 3.354-0.077 6.718-0.229 10.070-0.033 0.601-0.513 1.18-1.049 2.337z"  ></path><path d="M565.642 773.019c-0.164 28.618-21.507 49.436-50.277 49.054-28.235-0.371-49.567-22.25-49.206-50.463 0.35-27.307 22.523-48.322 50.561-47.929 28.399 0.394 49.087 21.266 48.923 49.338z"  ></path></symbol><symbol id="nsicon-loading" viewBox="0 0 1024 1024"><path d="M512 34.133333333333326q-14.9504-1.0581333333333334-28.808533333333333 12.8-4.266666666666667 4.266666666666667-6.929066666666667 11.195733333333333T473.6 68.26666666666665v4.266666666666667q0 16.008533333333332 9.591466666666667 25.6 5.3248 6.417066666666667 12.8 9.079466666666667t11.741866666666665 2.6624H512q108.81706666666668 0 201.59146666666666 53.8624t146.67093333333332 146.67093333333332 53.8624 202.1376-53.8624 201.59146666666666-146.67093333333332 146.1248-202.1376 53.8624-201.59146666666666-53.8624-146.1248-146.1248T109.87520000000006 512q0-16.008533333333332-11.741866666666665-27.7504-4.266666666666667-5.3248-11.195733333333333-7.987200000000001t-10.137599999999999-2.6624h-4.266666666666667q-16.008533333333332 0-26.658133333333335 10.683733333333333-6.417066666666667 6.417066666666667-9.079466666666667 13.346133333333334T34.133333333333326 507.7674666666666v4.266666666666667q0 97.0752 37.85386666666667 185.61706666666666t101.85386666666666 152.54186666666666 152.54186666666666 101.85386666666666T512 989.9008000000001t185.61706666666666-37.85386666666667 152.54186666666666-101.85386666666666 101.85386666666666-152.54186666666666T989.8666666666667 512.0341333333333t-37.85386666666667-185.61706666666666-101.85386666666666-152.54186666666666-152.54186666666666-101.85386666666666T512 34.167466666666655z"  ></path></symbol><symbol id="nsicon-yes" viewBox="0 0 1024 1024"><path d="M927.874 179.139l-553.426 553.426-280.413-280.254c-17.374-17.374-45.53-17.374-62.902 0s-17.374 45.53 0 62.902l343.316 343.155 616.329-616.329c17.374-17.374 17.374-45.53 0-62.902s-45.53-17.374-62.902 0z"  ></path></symbol><symbol id="nsicon-info" viewBox="0 0 1024 1024"><path d="M512 32C246.90399969 32 32 246.90399969 32 512s214.90399969 480 480 480 480-214.90399969 480-480S777.09600031 32 512 32z m30 709.99999969c0 16.5-13.5 30-30 30s-30-13.5-30-30V422c0-16.5 13.5-30 30-30s30 13.5 30 30v319.99999969z m-30-390c-22.092 0-40.00000031-17.90800031-40.00000031-39.99999938s17.90800031-40.00000031 40.00000031-40.00000031 40.00000031 17.90800031 40.00000031 40.00000031-17.90800031 40.00000031-40.00000031 39.99999938z"  ></path></symbol><symbol id="nsicon-no" viewBox="0 0 1024 1024"><path d="M568.4890625 509.1771875l392.59781250000003-392.59781250000003c16.947187500000002-16.947187500000002 16.947187500000002-42.365624999999994 0-59.3128125s-42.365624999999994-16.947187500000002-59.3128125 0l-392.59781250000003 392.59781250000003-392.59781250000003-392.59781250000003c-16.947187500000002-16.947187500000002-42.365624999999994-16.947187500000002-59.3128125 0s-16.947187500000002 42.365624999999994 0 59.3128125l392.59781250000003 392.59781250000003-392.59781250000003 392.59781250000003c-16.947187500000002 16.947187500000002-16.947187500000002 42.365624999999994 0 59.3128125 8.473125 8.473125 19.7709375 11.297812500000001 31.06875 11.297812500000001s22.5946875-2.8228125 31.06875-11.297812500000001l392.59781250000003-392.59781250000003 392.59781250000003 392.59781250000003c8.473125 8.473125 19.7709375 11.297812500000001 31.06875 11.297812500000001s22.5946875-2.8228125 31.06875-11.297812500000001c16.947187500000002-16.947187500000002 16.947187500000002-42.365624999999994 0-59.3128125l-398.24625-392.59781250000003z"  ></path></symbol><symbol id="nsicon-back" viewBox="0 0 1024 1024"><path d="M735.419 996.073c-9.93 0-19.859-2.482-27.307-9.93l-446.836-446.836c-14.895-14.895-14.895-37.236 0-52.131l446.836-446.836c14.895-14.895 37.236-14.895 52.131 0s14.895 37.236 0 52.131l-419.53 419.53 419.53 419.53c14.895 14.895 14.895 37.236 0 52.131-4.965 7.447-14.895 12.412-24.825 12.412z"  ></path></symbol><symbol id="nsicon-plus" viewBox="0 0 1024 1024"><path d="M1007.483871 469.058065h-462.451613v-462.451613h-66.064516v462.451613h-462.451613v66.064516h462.451613v462.451613h66.064516v-462.451613h462.451613z"  ></path></symbol><symbol id="nsicon-minus" viewBox="0 0 1024 1024"><path d="M1007.354 566.026l-990.981-0.023v-66.055h990.958z"  ></path></symbol><symbol id="nsicon-refresh" viewBox="0 0 1024 1024"><path d="M965.3333337500001 485.33333375c-16.0000003125 0-26.666666250000002 10.666666874999999-26.666667187500003 26.666666250000002 0 234.6666665625-192 426.6666665625-426.6666665625 426.6666665625s-426.6666665625-192-426.6666665625-426.6666665625 192-426.6666665625 426.6666665625-426.6666665625c141.33333375 0 271.9999996875 69.33333375000001 352.0000003125 184.00000031250002l2.66666625 2.66666625h-141.33333281249998c-16.0000003125 0-26.666666250000002 10.666666874999999-26.666667187500003 26.666666250000002s10.666666874999999 26.666666250000002 26.666667187500003 26.666667187500003h186.66666656249998c16.0000003125 0 26.666666250000002-10.666666874999999 26.666666250000002-26.666667187500003v-186.66666656249998c0-16.0000003125-10.666666874999999-26.666666250000002-26.666666250000002-26.666666250000002s-26.666666250000002 10.666666874999999-26.666667187500003 26.666666250000002v98.6666671875c-90.6666665625-112.0000003125-226.666666875-178.666666875-373.33333312499997-178.666666875-264 0-480 216-480 480s216 480 480 480 480-216 480-480c0-16.0000003125-10.666666874999999-26.666666250000002-26.666666250000002-26.666666250000002z"  ></path></symbol><symbol id="nsicon-error" viewBox="0 0 1024 1024"><path d="M512 32a480 480 0 1 0 0 960A480 480 0 1 0 512 32z m225.91999969 659.35999969c11.83999969 11.59999969 12 30.55999969 0.48 42.40000031a30.16000031 30.16000031 0 0 1-42.39999938 0.55999969L512 554 327.99999969 734.16000031a30 30 0 1 1-42-42.96L469.28 512l-183.19999969-179.35999969a30.16000031 30.16000031 0 0 1-0.40000031-42.40000031 30.16000031 30.16000031 0 0 1 42.40000031-0.55999969L512 470l184.00000031-180.16000031a30 30 0 1 1 42 42.96L554.72 512l183.19999969 179.35999969z"  ></path></symbol><symbol id="nsicon-right-arrow" viewBox="0 0 1024 1024"><path d="M288.58099999 27.927c9.93 0 19.859 2.482 27.30700001 9.93L762.724 484.693c14.895 14.895 14.895 37.236 0 52.131L315.888 983.66c-14.895 14.895-37.236 14.895-52.131 0s-14.895-37.236 0-52.131l419.53-419.53-419.53-419.53c-14.895-14.895-14.895-37.236 0-52.131 4.965-7.447 14.895-12.412 24.825-12.41200001z"  ></path></symbol></svg>';var script=function(){var scripts=document.getElementsByTagName("script");return scripts[scripts.length-1]}();var shouldInjectCss=script.getAttribute("data-injectcss");var ready=function(fn){if(document.addEventListener){if(~["complete","loaded","interactive"].indexOf(document.readyState)){setTimeout(fn,0)}else{var loadFn=function(){document.removeEventListener("DOMContentLoaded",loadFn,false);fn()};document.addEventListener("DOMContentLoaded",loadFn,false)}}else if(document.attachEvent){IEContentLoaded(window,fn)}function IEContentLoaded(w,fn){var d=w.document,done=false,init=function(){if(!done){done=true;fn()}};var polling=function(){try{d.documentElement.doScroll("left")}catch(e){setTimeout(polling,50);return}init()};polling();d.onreadystatechange=function(){if(d.readyState=="complete"){d.onreadystatechange=null;init()}}}};var before=function(el,target){target.parentNode.insertBefore(el,target)};var prepend=function(el,target){if(target.firstChild){before(el,target.firstChild)}else{target.appendChild(el)}};function appendSvg(){var div,svg;div=document.createElement("div");div.innerHTML=svgSprite;svgSprite=null;svg=div.getElementsByTagName("svg")[0];if(svg){svg.setAttribute("aria-hidden","true");svg.style.position="absolute";svg.style.width=0;svg.style.height=0;svg.style.overflow="hidden";prepend(svg,document.body)}}if(shouldInjectCss&&!window.__iconfont__svg__cssinject__){window.__iconfont__svg__cssinject__=true;try{document.write("<style>.svgfont {display: inline-block;width: 1em;height: 1em;fill: currentColor;vertical-align: -0.1em;font-size:16px;}</style>")}catch(e){console&&console.log(e)}}ready(appendSvg)})(window)
	
}