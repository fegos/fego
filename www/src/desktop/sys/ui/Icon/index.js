import React, { Component } from 'react';
import { Icon } from 'fego';
import style from './index.less';

const icons = [
  'question-fill',
  'error',
  'info',
  'warning',
  'success',
  'home',
  'minus',
  'plus',
  'yes',
  'no',
];

export default class Page extends Component {
  state = {}
  render() {
    return (
      <div className={style.page}>
        <div className="section f-cb">
          {icons.map(icon => (
            <div className="item" key={icon}>
              <Icon name={icon} />
              <span className={style.name}>{icon}</span>
            </div>
          ))}
        </div>
      </div>
    );
  }
}
