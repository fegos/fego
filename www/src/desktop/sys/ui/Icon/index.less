.page {
  :global {
    .item {
      width: 100px;
      height: 100px;
      float: left;
      margin-right: 20px;
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      span {
        margin-top: 15px;
      }
    }
  }
}
