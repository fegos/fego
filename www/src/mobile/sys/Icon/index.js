import React, { Component } from 'react'
import { Icon } from 'fego'
import style from './index.less'

const icons = [
	'question-fill',
	'error',
	'info',
	'warning',
	'success',
	'home',
	'minus',
	'plus',
	'yes',
	'no',
]

export default class Page extends Component {
	render() {
		return (
			<div className={style.page}>
			{icons.map(icon => {
				return (
					<div className={style.item} key={icon}>
						<Icon name={icon} />
						<span className={style.name}>{icon}</span>
					</div>
				)
			})}
			</div>
		)
	}
}