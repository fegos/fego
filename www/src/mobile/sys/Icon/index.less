.page {
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	.item {
		display: inline-block;
		width: 25%;
		text-align: center;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		height: 80px;
		.name {
			font-size: 14px;
			margin-top: 10px;
		}
	}
}