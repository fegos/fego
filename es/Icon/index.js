'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = Icon;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('./font.js');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                              * 图标组件
                                                                                                                                                                                                                              * @author esky
                                                                                                                                                                                                                              */


function Icon(props) {
  var name = props.name,
      _props$className = props.className,
      className = _props$className === undefined ? '' : _props$className,
      spin = props.spin,
      size = props.size,
      width = props.width,
      height = props.height,
      _props$hrefPrefix = props.hrefPrefix,
      hrefPrefix = _props$hrefPrefix === undefined ? 'nsicon' : _props$hrefPrefix,
      style = props.style,
      rest = _objectWithoutProperties(props, ['name', 'className', 'spin', 'size', 'width', 'height', 'hrefPrefix', 'style']);

  var cls = (className + ' nsicon ' + (spin || name === 'loading' ? 'nsicon-spin' : '')).trim();
  var _style = _extends({}, style, { width: size || width, height: size || height });
  /**
  <!-- REACT JSX: -->
  <svg> <use xlinkHref='/svg/svg-sprite#my-icon' /> </svg>
  <!-- RENDERS AS: -->
  <svg> <use xlink:href="/svg/svg-sprite#my-icon"></use> </svg>
    */
  return _react2.default.createElement(
    'svg',
    _extends({ className: cls, style: _style, 'aria-hidden': 'true' }, rest),
    _react2.default.createElement('use', { xlinkHref: '#' + hrefPrefix + '-' + name })
  );
}
/**
symbol引用

这是一种全新的使用方式，应该说这才是未来的主流，也是平台目前推荐的用法。相关介绍可以参考这篇文章 这种用法其实是做了一个svg的集合，与另外两种相比具有如下特点：

支持多色图标了，不再受单色限制。
通过一些技巧，支持像字体那样，通过font-size,color来调整样式。
兼容性较差，支持 ie9+,及现代浏览器。
浏览器渲染svg的性能一般，还不如png。
 */