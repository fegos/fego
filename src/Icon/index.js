/**
 * 图标组件
 * @author esky
 */
import React from 'react'
import './font'
export default function Icon(props) {
  const { name, className = '', spin, size, width, height, hrefPrefix = 'nsicon', style, ...rest } = props;
  const cls = `${className} nsicon ${spin || name === 'loading' ? 'nsicon-spin' : ''}`.trim();
  let _style = { ...style, width: size || width, height: size || height };
  /**
  <!-- REACT JSX: -->
  <svg> <use xlinkHref='/svg/svg-sprite#my-icon' /> </svg>
  <!-- RENDERS AS: -->
  <svg> <use xlink:href="/svg/svg-sprite#my-icon"></use> </svg>
    */
  return <svg className={cls} style={_style} aria-hidden="true" {...rest}><use xlinkHref={`#${hrefPrefix}-${name}`} /></svg>
}
/**
symbol引用

这是一种全新的使用方式，应该说这才是未来的主流，也是平台目前推荐的用法。相关介绍可以参考这篇文章 这种用法其实是做了一个svg的集合，与另外两种相比具有如下特点：

支持多色图标了，不再受单色限制。
通过一些技巧，支持像字体那样，通过font-size,color来调整样式。
兼容性较差，支持 ie9+,及现代浏览器。
浏览器渲染svg的性能一般，还不如png。
 */
