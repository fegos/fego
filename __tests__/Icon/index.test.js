import React from 'react';
import Renderer from 'react-test-renderer';
import Icon from 'Icon';

describe('Icon test', () => {
  it('icon snapshot', () => {
    const icon = Renderer.create(<Icon name="yes" />);
    expect(icon.toJSON()).toMatchSnapshot();
  });
});
