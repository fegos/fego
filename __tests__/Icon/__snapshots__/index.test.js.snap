// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test icon snapshot 1`] = `
<svg
  aria-hidden="true"
  className="nsicon"
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <use
    xlinkHref="#nsicon-yes"
  />
</svg>
`;
